File: container-images/admiral-agent/go/src/shellserver/dockerutils.go
/*
 * Copyright (c) 2016 VMware, Inc. All Rights Reserved.
 *
 * This product is licensed to you under the Apache License, Version 2.0 (the "License").
 * You may not use this product except in compliance with the License.
 *
 * This product may include a number of subcomponents with separate copyright notices
 * and license terms. Your use of these subcomponents is subject to the terms and
 * conditions of the subcomponent's license, as noted in the LICENSE file.
 */

package main

import (
	"github.com/fsouza/go-dockerclient"
)

var endpoint = "unix:///var/run/docker.sock"
var client, _ = docker.NewClient(endpoint)

func IsContainerRunning(containerNameOrId string) (bool, error) {
	var container, err = client.InspectContainer(containerNameOrId)
	if err != nil {
		return false, err
	}

	return container.State.Running, nil
}

func GetContainerId(containerNameOrId string) (string, error) {
	var container, err = client.InspectContainer(containerNameOrId)
	if err != nil {
		return "", err
	}
	return container.ID, nil
}

File: test-e2e/e2e-test-common/src/main/java/com/vmware/admiral/test/util/DockerUtils.java
/*
 * Copyright (c) 2018 VMware, Inc. All Rights Reserved.
 *
 * This product is licensed to you under the Apache License, Version 2.0 (the "License").
 * You may not use this product except in compliance with the License.
 *
 * This product may include a number of subcomponents with separate copyright notices
 * and license terms. Your use of these subcomponents is subject to the terms and
 * conditions of the subcomponent's license, as noted in the LICENSE file.
 */

package com.vmware.admiral.test.util;

import javax.net.ssl.HostnameVerifier;
import javax.net.ssl.SSLContext;

import com.spotify.docker.client.DefaultDockerClient;
import com.spotify.docker.client.DockerCertificatesStore;
import com.spotify.docker.client.DockerClient;

public class DockerUtils {

    public static DockerClient createUnsecureDockerClient(String uri) {

        DockerCertificatesStore store = new DockerCertificatesStore() {

            @Override
            public SSLContext sslContext() {
                return HttpUtils.newUnsecureSSLContext();
            }

            @Override
            public HostnameVerifier hostnameVerifier() {
                return HttpUtils.allowAllHostNameVeririer();
            }
        };
        return DefaultDockerClient.builder()
                .uri(uri)
                .dockerCertificates(store)
                .readTimeoutMillis(120000)
                .connectTimeoutMillis(120000)
                .build();
    }

}
